app/controllers/cards_controller.rb

```diff
@@ -15,39 +16,35 @@ def create
           requires(:holder_type).one_of('Supporter')
         end
       end.when_valid do |d|
-        ret = nil
-        recaptcha_result = check_recaptcha(action: 'create_card', minimum_score: ENV['MINIMUM_RECAPTCHA_SCORE'].to_f)
-        if !recaptcha_result[:success]
-
-          failure_details = {
-            supporter: supporter.attributes,
-            validated_params: d,
-            action: 'create_card',
-            minimum_score_required: ENV['MINIMUM_RECAPTCHA_SCORE'],
-            recaptcha_result: recaptcha_result,
-            recaptcha_value: d['g-recaptcha-response']
-          }
-          failure = RecaptchaRejection.new
-          failure.details_json = failure_details
-          failure.save!
-          ret = {json: {error: "There was an temporary error preventing your payment. Please try again. If it persists, please contact support@commitchange.com with error code: 5X4J "}, status: :unprocessable_entity}
-        end
-        unless ret
-          if (recaptcha_result[:reply] && recaptcha_result[:reply]['success'])
-            # Rails.logger({supporter: supporter.attributes,
-            #   validated_params: d,
-            #   action: 'create_card',
-            #   minimum_score_required: ENV['MINIMUM_RECAPTCHA_SCORE'],
-            #   recaptcha_result: recaptcha_result,
-            #   recaptcha_value: d['g-recaptcha-response']})
-            ret = InsertCard.with_stripe(d[:card], acct,  params[:event_id], current_user)
-          else
-            ret = {json: {error: "There was an temporary error preventing your payment. Please try again. If it persists, please contact support@commitchange.com with error code: 5X4J "}, status: :unprocessable_entity}
-          end
-        end
-        ret
+        supporter = Supporter.find(d[:card][:holder_id])
+        acct = supporter.nonprofit.stripe_account_id
+        InsertCard.with_stripe(d[:card], acct,  params[:event_id], current_user)
       end
     )
 	end
 
+  private
+  def verify_via_recaptcha!
+    begin
+      verify_recaptcha!(action: 'create_card', minimum_score: ENV['MINIMUM_RECAPTCHA_SCORE'].to_f)
+    rescue ::Recaptcha::RecaptchaError => e
+      supporter_id = params.try(:card).try(:holder_id)
+      failure_details = {
+        supporter: supporter_id,
+        params: params,
+        action: 'create_card',
+        minimum_score_required: ENV['MINIMUM_RECAPTCHA_SCORE'],
+        recaptcha_result: recaptcha_reply,
+        recaptcha_value: params['g-recaptcha-response']
+      }
+      failure = RecaptchaRejection.new
+      failure.details_json = failure_details
+      failure.save!
+      raise e
+    end
+  end
```

Comment:
I move the recaptcha check into a separate callback to pull it out of the action itself. It simplifies the `/create` action